crates/sui-types/src/gas.rs

```diff
@@ -60,7 +60,7 @@ impl<'a> SuiGasStatus<'a> {
                 gas_price,
                 config,
             )),
-            2 => Self::V2(SuiGasStatusV2::new_with_budget(
+            2 | 3 => Self::V2(SuiGasStatusV2::new_with_budget(
```

Comment:
There are quite a few places where we match on gas model version, which is an u64. This can be hard to maintain down the road. I recommend we either introduce feature flags for it (so that we centralize checks inside protocol config), or we turn the gas model number into an enum so that we always remember to update all places. Non-blocking though

Reply:
I'll look into that, funny enough I thought about feature flag and thought exactly the opposite of what you are thinking. I felt they would be incredibly difficult to reconcile once they explode. I liked the sequential grow of version and having everything gated behind a single number.
In any case I will think about it and play devil's advocate against myself to see if I can convince myself a feature flag would be a better way to go.
Thanks @lxfind

Reply:
It doesn't need to be an actual feature flag. The important piece here is that we don't expose gas model number explicitly and thus the caller has to check the number manually.
For instance, it would already be a lot easier to manage if we introduce the following APIs in the protocol config (with better names):
fn supports_gas_mode_v2() -> bool { self.gas_model_number >= 2 }
fn fixes_input_object_version_check_in_gas() -> bool { self.gas_model_number >= 3 }
and etc.
That way you don't have to expand every match on gas model version each time a new version is added.